Pass TARGET_CONFIG and HOST_CONFIG from SPEC file to inner configure invoked
by setup.py to cross compile internal libffi required by _ctypes.so

Signed-off-by: Fabio Arnone <fabio.arnone@st.com>
Reviewed-by: Carmelo Amoroso <carmelo.amoroso@st.com>

--- Python-2.6.2/setup.py.orig	2011-01-03 11:44:32.000000000 +0100
+++ Python-2.6.2/setup.py	2011-01-03 11:46:24.000000000 +0100
@@ -1698,7 +1698,7 @@ class PyBuildExt(build_ext):
                                          ffi_configfile):
                 from distutils.dir_util import mkpath
                 mkpath(ffi_builddir)
-                config_args = ['--host=%s' % os.environ["TARGET_OS"], ]
+                config_args = ['--target=' + str(os.environ.get('TARGET_CONFIG'))  + ' --host=' + str(os.environ.get('HOST_CONFIG')), ]
 
                 # Pass empty CFLAGS because we'll just append the resulting
                 # CFLAGS to Python's; -g or -O2 is to be avoided.
